genrl/deep/agents/td3/td3.py

```diff
@@ -119,25 +126,37 @@ def __init__(
         self.empty_logs()
         self.create_model()
 
-    def create_model(self) -> None:
-        state_dim, action_dim, discrete, _ = get_env_properties(self.env)
-        if discrete:
-            raise Exception(
-                "Discrete Environments not supported for {}.".format(__class__.__name__)
+    def create_model(self, **kwargs) -> None:
+        if isinstance(self.network, str):
+            state_dim, action_dim, discrete, _ = get_env_properties(self.env)
+            assert not discrete, "Discrete Environments not supported for {}.".format(
+                __class__.__name__
             )
+
+            self.ac = get_model("ac", self.network)(
+                state_dim, action_dim, self.layers, "Qsa", False
+            ).to(self.device)
+
+            self.ac.qf2 = get_model("v", self.network)(
+                state_dim, action_dim, hidden=self.layers, val_type="Qsa"
+            )
+        else:
+            self.ac = self.network.to(self.device)
+            action_dim = self.network.action_dim
+            assert (
+                "actor" and "critic" and "value" in dir(self.ac)
+            ), "network must contain actor, critic and value attributes"
+            assert "get_action" and "get_value" in dir(
+                self.ac
+            ), "network must contain get_action and get_action methods"
+            self.ac.qf2 = self.ac.value

```

Comment:
This is again like SAC. It's originally supposed to be `qf1` and `qf2`. Refer to #190.

You'll need to remove "value from the assert statement. Also, `self.ac.qf2 = self.ac.value` would again be wrong because that would make it identical. For theoretical clarification, you can refer to the original SAC and TD3 papers.